public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -96,12 +97,25 @@ export class GrafanaDatasource extends DataSourceWithBackend<GrafanaQuery> {
       if (target.queryType === GrafanaQueryType.Snapshot) {
         results.push(
           of({
+            // NOTE refId is intentionally missing because:
+            // 1) there is only one snapshot
+            // 2) the payload will reference original refIds
             data: (target.snapshot ?? []).map((v) => dataFrameFromJSON(v)),
             state: LoadingState.Done,
           })
         );
         continue;
       }
+      if (target.queryType === GrafanaQueryType.TimeRegions) {
+        const frame = doTimeRegionQuery('', target.timeRegion!, request.range, request.timezone);
```

Comment:
```suggestion
        const frame = doTimeRegionQuery('', target.timeRegion ?? defaultTimeRegionHere, request.range, request.timezone);
```

Does it make sense for timeRegion to be an optional property? 🤔 Seems like we are referencing this in multiple places assuming it should be defined (also see below on line 198)

Reply:
For `timeRegions` yes, since it's defined in `GrafanaQuery` and other query types won't have it. But for a time region query I think it'll always be defined. Not so sure about `timezone` though, you can't clear it in the UI and we're defaulting to dashboard timezone on migration, so actually I don't think that will ever be undefined. But you never know?